src/Registration/cReg/NiftiImageData3DBSpline.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+SyneRBI Synergistic Image Reconstruction Framework (SIRF)
+Copyright 2017 - 2020 University College London
+
+This is software developed for the Collaborative Computational
+Project in Synergistic Reconstruction for Biomedical Imaging (formerly CCP PETMR)
+(http://www.ccpsynerbi.ac.uk/).
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+/*!
+\file
+\ingroup Registration
+\brief Class for deformation field transformations.
+
+\author Richard Brown
+\author SyneRBI
+*/
+
+#include "sirf/Reg/NiftiImageData3DBSpline.h"
+#include "sirf/Reg/NiftiImageData3DDeformation.h"
+#include "sirf/NiftyMoMo/BSplineTransformation.h"
+
+using namespace sirf;
+
+template<class dataType>
+NiftiImageData3DBSpline<dataType>::NiftiImageData3DBSpline(const NiftiImageData3DDeformation<dataType> &def, const float spacing[])
```

Comment:
this is supposed to be the implementation of the `ControlPointGridToDeformationConverter::backward`. I wouldn't put it in the this constructor, but keep it there. Most people would expect this to do something smart with finding the best CPG that fits the `def`, but that's not what `backward` is supposed to do I believe.

Think about this is way: there's some matrix `B` corresponding to `dvf = converter.forward(cpg) = B cpg`. The backward operation is supposed to be `cpgnew = converter.backward(dvf) = B' cpg`. So, if we'd do them in sequence `cpgnew = B' B cpg`. There is no reason to believe that `B'B` is an identity matrix.

Reply:
constructor removed (and placed in `backward`).